pkg/api/provision.go

```diff
@@ -395,13 +406,30 @@ func (s *server) provision(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	task := model.NewTask(
-		"provisionStep",
-		map[string]string{
-			"stepName":   firstStepName,
-			"instanceID": instanceID,
-		},
-	)
+	var task model.Task
+	if waitForParent {
+		task = model.NewDelayedTask(
+			"waitForParentStep",
+			map[string]string{
+				"provisionFirstStep": firstStepName,
```

Comment:
Same as [this comment](https://github.com/Azure/open-service-broker-azure/pull/204/files#r161847903)

Reply:
Same as [this comment](https://github.com/Azure/open-service-broker-azure/pull/204/files#r161852255)